server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -922,70 +924,129 @@ private static void validateTemplate(Settings validateSettings, String mappings,
         }
     }
 
-    private void validate(PutRequest request) {
+    private void validate(String name, ComponentTemplate template) {
```

Comment:
These two methods are nearly identical so they can be refactor to avoid duplication
```java
private void validate(String name, ComponentTemplate template) {
    validate(name, template.template(), Collections.emptyList());
}

private void validate(String name, IndexTemplateV2 template) {
    validate(name, template.template(), template.indexPatterns());
}

private void validate(String name, Template template1, List<String> indexPatterns) {
    Optional<Template> maybeTemplate = Optional.ofNullable(template1);
    validate(name,
        maybeTemplate.map(Template::settings).orElse(Settings.EMPTY),
        indexPatterns,
        maybeTemplate.map(Template::aliases)
            .orElse(Collections.emptyMap())
            .values().stream()
            .map(this::toAlias)
            .collect(Collectors.toList()));
}

private Alias toAlias(AliasMetadata aliasMeta) {
    Alias a = new Alias(aliasMeta.alias());
    if (aliasMeta.filter() != null) {
        a.filter(aliasMeta.filter().string());
    }
    a.searchRouting(aliasMeta.searchRouting());
    a.indexRouting(aliasMeta.indexRouting());
    a.isHidden(aliasMeta.isHidden());
    a.writeIndex(aliasMeta.writeIndex());
    return a;
}
```

Reply:
Good call, did this, thanks!